src/precice/impl/SolverInterfaceImpl.cpp

```diff
@@ -1689,10 +1666,7 @@ void SolverInterfaceImpl::syncTimestep(double computedTimestepLength)
 const mesh::Mesh &SolverInterfaceImpl::mesh(const std::string &meshName) const
 {
   PRECICE_TRACE(meshName);
-  const MeshContext *context = _accessor->usedMeshContextByName(meshName);
-  PRECICE_ASSERT(context && context->mesh,
-                 "Participant \"" << _accessorName << "\" does not use mesh \"" << meshName << "\"!");
-  return *context->mesh;
+  return *_accessor->usedMeshContext(meshName).mesh;
```

Comment:
Does it actually resolve #989 ?

Reply:
This function is used for tests only and not part of the SolverInterface.
#989 should be fixed with the changes in the ValidationMacros

Reply:
I guess I was in the wrong line when adding my review comment here. The `DataContext` call is gone in the respective validation macro, so I guess the PR still resolves the issue in #989. Let me try it out once this has been merged.